backend/mr-db-sql/src/main/kotlin/io/featurehub/db/services/FeatureSqlApi.kt

```diff
@@ -196,17 +196,21 @@ class FeatureSqlApi @Inject constructor(
   ) {
     val feature = existing.feature
 
-    val lockChanged = updateSelectivelyLocked(featureValue, historical, existing, person)
+    val lockUpdate = updateSelectivelyLocked(featureValue, historical, existing, person)
+    val lockChanged = lockUpdate.hasChanged
 
     // allow them to change the value and lock it at the same time
-    val defaultValueChanged =
+    val defaultValueUpdate =
       updateSelectivelyDefaultValue(feature, featureValue, historical, existing, person, lockChanged)
+    val defaultValueChanged = defaultValueUpdate.hasChanged
 
-    val strategiesChanged = updateSelectivelyRolloutStrategies(person, featureValue, historical, existing, lockChanged)
+    val strategyUpdates = updateSelectivelyRolloutStrategies(person, featureValue, historical, existing, lockChanged)
+    val strategiesChanged = strategyUpdates.hasChanged
 
-    val retiredChange = updateSelectivelyRetired(person, featureValue, historical, existing, lockChanged)
+    val retiredUpdate = updateSelectivelyRetired(person, featureValue, historical, existing, lockChanged)
+    val retiredChanged = retiredUpdate.hasChanged
 
-    if (lockChanged || defaultValueChanged || strategiesChanged || retiredChange) {
+    if (lockChanged || defaultValueChanged || strategiesChanged || retiredChanged) {
```

Comment:
you probably don't need to create the variables, could just be lockUpdate.hasChanged, etc?

Reply:
fixed